enterprise/fulltext-addon/src/test/java/org/neo4j/kernel/api/impl/fulltext/LuceneFulltextUpdaterTest.java

```diff
@@ -666,17 +686,85 @@ public void shouldNotReturnNonMatches() throws Exception
 
                 tx.success();
             }
-            try ( ReadOnlyFulltext reader = provider.getReader( "nodes", FulltextIndexType.NODES ) )
+            try ( ReadOnlyFulltext reader = provider.getReader( "nodes", NODES ) )
             {
                 PrimitiveLongIterator zebra = reader.query( "zebra" );
                 assertFalse( zebra.hasNext() );
             }
-            try ( ReadOnlyFulltext reader = provider.getReader( "relationships", FulltextIndexType.RELATIONSHIPS ) )
+            try ( ReadOnlyFulltext reader = provider.getReader( "relationships", RELATIONSHIPS ) )
             {
 
                 PrimitiveLongIterator zebra = reader.query( "zebra" );
                 assertFalse( zebra.hasNext() );
             }
         }
     }
+
+    @Test
+    public void shouldPopulateIndexWithExistingNodesAndRelationships() throws Exception
+    {
+        GraphDatabaseAPI db = dbRule.getGraphDatabaseAPI();
+        FulltextFactory fulltextFactory = new FulltextFactory( fileSystemRule, testDirectory.graphDbDir(), ANALYZER );
+
+        long firstNodeID;
+        long secondNodeID;
+        long firstRelID;
+        long secondRelID;
+        try ( Transaction tx = db.beginTx() )
+        {
+            Node node = db.createNode( LABEL );
+            Node node2 = db.createNode( LABEL );
+            Relationship ignore1 = node.createRelationshipTo( node2, RELTYPE );
+            Relationship ignore2 = node.createRelationshipTo( node2, RELTYPE );
+            Relationship rel1 = node.createRelationshipTo( node2, RELTYPE );
+            Relationship rel2 = node2.createRelationshipTo( node, RELTYPE );
+            firstNodeID = node.getId();
+            secondNodeID = node2.getId();
+            firstRelID = rel1.getId();
+            secondRelID = rel2.getId();
+            node.setProperty( "prop", "Hello. Hello again." );
+            node2.setProperty( "prop", "This string is slightly shorter than the zebra one" );
+            rel1.setProperty( "prop", "Goodbye" );
+            rel2.setProperty( "prop", "And now, something completely different" );
+
+            tx.success();
+        }
+
+        try ( FulltextProvider provider = new FulltextProvider( db, LOG, availabilityGuard ) )
+        {
+            fulltextFactory.createFulltextIndex( "nodes", NODES, singletonList( "prop" ), provider );
+            fulltextFactory.createFulltextIndex( "relationships", RELATIONSHIPS, singletonList( "prop" ), provider );
+            provider.init();
+            provider.awaitPopulation();
+
+            try ( ReadOnlyFulltext reader = provider.getReader( "nodes", NODES ) )
+            {
+
+                PrimitiveLongIterator hello = reader.query( "hello" );
```

Comment:
it's super hard to understand what this test is actually testing because of zebras of asserts, can those be grouped extracted in some way to be able to understand whats going on without  👓  and 🍷

Reply:
Let's do that in #10079 where we've already made significant changes to this file.